script.js

```diff
@@ -0,0 +1,682 @@
+                            if (keyArray[i][j].letterEn === 1) {
+                                textareaSelector.value = textareaSelector.value.substring(0, textareaSelector.selectionStart) + keyArray[i][j].EN + textareaSelector.value.substring(textareaSelector.selectionEnd);
+                            } else {
+                                textareaSelector.value = textareaSelector.value.substring(0, textareaSelector.selectionStart) + keyArray[i][j].en + textareaSelector.value.substring(textareaSelector.selectionEnd);
+                            }
+                            textareaSelector.selectionEnd = start + 1;
+                        }
+                        break;
+                    case 'RU':
+                        if (keyArray[i][j].activeKey === 0) {
+                            const start = textareaSelector.selectionStart;
+                            if (keyArray[i][j].letterRu === 1) {
+                                textareaSelector.value = textareaSelector.value.substring(0, textareaSelector.selectionStart) + keyArray[i][j].RU + textareaSelector.value.substring(textareaSelector.selectionEnd);
+                            } else {
+                                textareaSelector.value = textareaSelector.value.substring(0, textareaSelector.selectionStart) + keyArray[i][j].ru + textareaSelector.value.substring(textareaSelector.selectionEnd);
+                            }
+                            textareaSelector.selectionEnd = start + 1;
+                        }
+                        break;
+                    case 'ru':
+                        if (keyArray[i][j].activeKey === 0) {
+                            const start = textareaSelector.selectionStart;
+                            textareaSelector.value = textareaSelector.value.substring(0, textareaSelector.selectionStart) + keyArray[i][j].ru + textareaSelector.value.substring(textareaSelector.selectionEnd);
+                            textareaSelector.selectionEnd = start + 1;
+                        }
+                        break;
+                }
+            }
+        }
+    }
+});
+
+document.addEventListener('mouseup', (event) => {
+    for (let i = 0; i < keyArray.length; i++) {
+        for (let j = 0; j < keyArray[i].length; j++) {
+            if (event.target.classList.contains(keyArray[i][j].code)) {
+                const pressButton = document.querySelector(`.${keyArray[i][j].code}`);
+                pressButton.classList.remove('press');
+            }
+        }
+    }
+});
+
+// make alt work correctly
+document.addEventListener('keydown', (event) => {
+  if (event.code === 'AltLeft' || event.code === 'AltRight') {
+    event.preventDefault();
+  }
+});
+
+// make tab work correctly
+const textareaSelector = document.querySelector('.textarea');
+textareaSelector.onkeydown = function(event) {
+  if (event.code === 'Tab') {
+    event.preventDefault();
+    var s = this.selectionStart;
```

Comment:
be consistent, use either `let, const` or `var`
use proper names for variables

you should use `textareaSelector` here, not `this` + you have `event.target`